src/Transports/MassTransit.EventHubIntegration/EventHubIntegration/Checkpoints/PendingConfirmationCollection.cs

```diff
@@ -9,17 +9,17 @@ namespace MassTransit.EventHubIntegration.Checkpoints
     public class PendingConfirmationCollection :
         IDisposable
     {
-        readonly ConcurrentDictionary<long, IPendingConfirmation> _confirmations;
-        readonly string _eventHubName;
+        readonly CancellationToken _cancellationToken;
+        readonly ConcurrentDictionary<PartitionOffset, IPendingConfirmation> _confirmations;
         readonly CancellationTokenRegistration? _registration;
 
-        public PendingConfirmationCollection(string eventHubName, CancellationToken cancellationToken)
+        public PendingConfirmationCollection(CancellationToken cancellationToken)
         {
-            _eventHubName = eventHubName;
-            _confirmations = new ConcurrentDictionary<long, IPendingConfirmation>();
+            _cancellationToken = cancellationToken;
+            _confirmations = new ConcurrentDictionary<PartitionOffset, IPendingConfirmation>();
 
-            if (cancellationToken.CanBeCanceled)
-                _registration = cancellationToken.Register(() => Cancel(cancellationToken));
+            if (_cancellationToken.CanBeCanceled)
```

Comment:
General guidance is to use parameters in constructors, instead of properties. AKA, `cancellationToken` vs. `_can...`